pkg/reconciler/service/resources/configuration.go

```diff
@@ -17,31 +17,51 @@ limitations under the License.
 package resources
 
 import (
+	"strings"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"knative.dev/pkg/kmeta"
+	cfgmap "knative.dev/serving/pkg/apis/config"
 	"knative.dev/serving/pkg/apis/serving"
 	v1 "knative.dev/serving/pkg/apis/serving/v1"
 	"knative.dev/serving/pkg/reconciler/service/resources/names"
 )
 
 // MakeConfiguration creates a Configuration from a Service object.
 func MakeConfiguration(service *v1.Service) (*v1.Configuration, error) {
+	return MakeConfigurationFromExisting(service, &v1.Configuration{}, cfgmap.Disabled)
+}
+
+func MakeConfigurationFromExisting(service *v1.Service, existing *v1.Configuration, gc cfgmap.Flag) (*v1.Configuration, error) {
+	labels := map[string]string{serving.ServiceLabelKey: service.Name}
+	anns := kmeta.FilterMap(service.GetAnnotations(), func(key string) bool {
+		return key == corev1.LastAppliedConfigAnnotation
+	})
+
+	routeName := names.Route(service)
+	if gc != cfgmap.Enabled {
+		labels[serving.RouteLabelKey] = routeName
+	}
+
+	if gc == cfgmap.Enabled || gc == cfgmap.Allowed {
+		if val, has := existing.Annotations[serving.RoutesAnnotationKey]; has && strings.Contains(val, routeName) {
```

Comment:
probably can ignore the check for `has`
```suggestion
		if val:= existing.Annotations[serving.RoutesAnnotationKey]; strings.Contains(val, routeName) {
```

Reply:
But I also don't think this is a correct check. routeName = "ab" will math "q,abc,r", but it's not there.